helix-tui/src/terminal.rs

```diff
@@ -138,11 +138,9 @@ where
 
     /// Queries the backend for size and resizes if it doesn't match the previous size.
     pub fn autoresize(&mut self) -> io::Result<()> {
-        if self.viewport.resize_behavior == ResizeBehavior::Auto {
-            let size = self.size()?;
-            if size != self.viewport.area {
-                self.resize(size)?;
-            }
+        let size = self.size()?;
+        if size != self.viewport.area {
+            self.resize(size)?;
         };
         Ok(())
```

Comment:
Why not return `size` here so we don't need to `self.size()` again in compositor?